src/main/java/ch/njol/skript/expressions/ExprFunction.java

```diff
@@ -0,0 +1,131 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.ScriptLoader;
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.function.DynamicFunctionReference;
+import ch.njol.skript.lang.function.Functions;
+import ch.njol.skript.lang.function.Namespace;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.registrations.Feature;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.lang.script.Script;
+
+import java.io.File;
+import java.util.Objects;
+
+@Name("Function (Experimental)")
+@Description("Obtain a function by name, which can be executed.")
+@Examples({
+		"set {_function} to the function named \"myFunction\"",
+		"run {_function} with arguments 13 and true"
+})
+@Since("INSERT VERSION")
+@SuppressWarnings("rawtypes")
+public class ExprFunction extends SimpleExpression<DynamicFunctionReference> {
+
+	static {
+		Skript.registerExpression(ExprFunction.class, DynamicFunctionReference.class, ExpressionType.COMBINED,
+				"[the|a] function [named] %string% [local:(in|from) %-script%]",
+				"[the] functions [named] %strings% [local:(in|from) %-script%]",
+				"[all [[of] the]|the] functions (in|from) %script%"
+		);
+	}
+
+	private Expression<String> name;
```

Comment:
Should be annotated as nullable probably

Reply:
No, the Nullable annotation is for when we have no way to determine whether something is null.
In this case, it being null is contingent on the matched pattern (`mode` field).
<img width="649" alt="image" src="https://github.com/user-attachments/assets/308d5bd8-0c50-4e2c-a912-566113cdcd75" />